src/openssh/patch/0004-ssh-crash-fix.patch

```diff
@@ -0,0 +1,107 @@
+From 745b435a3bf705987ed546e32bd9fe0c6e5378f3 Mon Sep 17 00:00:00 2001
+From: Asha Behera <asha.behera@broadcom.com>
+Date: Mon, 24 Feb 2025 23:19:10 -0700
+Subject: [PATCH] [PATCH] ssh crash fix
+
+---
+ auth.c    | 9 +++++++--
+ auth.h    | 2 +-
+ packet.c  | 2 +-
+ session.c | 5 ++++-
+ sshd.c    | 6 ++++--
+ 5 files changed, 17 insertions(+), 7 deletions(-)
+
+diff --git a/auth.c b/auth.c
+index 73a85fd..4b4a7b2 100644
+--- a/auth.c
++++ b/auth.c
+@@ -1191,12 +1191,17 @@ auth_authorise_keyopts(struct ssh *ssh, struct passwd *pw,
+ }
+ 
+ /* Export remote IP address and port for authorization. */
+-void
++int
+ export_remote_info(struct ssh *ssh)
+ {
++        if (ssh == NULL)
```

Comment:
@liuh-80 If we do not intend to do anything with return status of export_remote_info as it will change behavior ...I will chnage as below :   Hope it is ok ?
/* Export remote IP address and port for authorization. */
void
export_remote_info(struct ssh *ssh)
{
        if (ssh != NULL)
		{
			const char *remote_ip = ssh_remote_ipaddr(ssh);
			const int remote_port = ssh_remote_port(ssh);
			const char remote_addr_port[32 + INET6_ADDRSTRLEN];
			snprintf(remote_addr_port, sizeof(remote_addr_port), "%s %d", remote_ip, remote_port);
			setenv("SSH_CLIENT_IPADDR_PORT", remote_addr_port, 1);
		}
}